tests/ImageSharp.Tests/ImageComparer.cs

```diff
@@ -0,0 +1,119 @@
+﻿namespace ImageSharp.Tests
+{
+    using System;
+    using ImageSharp;
+    using Xunit;
+
+    /// <summary>
+    /// Class to perform simple image comparisons.
+    /// </summary>
+    public static class ImageComparer
+    {
+        const int DefaultScalingFactor = 32; // this is means the images get scaled into a 32x32 image to sample pixels
+        const int DefaultSegmentThreshold = 3; // the greyscale difference between 2 segements my be > 3 before it influances the overall difference
+        const float DefaultImageThreshold = 0.000f; // after segment threasholds the images must have no differences
+
+        /// <summary>
+        /// Does a visual comparison between 2 images and then asserts the difference is less then a configurable threshold
+        /// </summary>
+        /// <typeparam name="TColorA">The color of the expected image</typeparam>
+        /// <typeparam name="TColorB">The color type fo the the actual image</typeparam>
+        /// <param name="expected">The expected image</param>
+        /// <param name="actual">The actual image</param>
+        /// <param name="imageTheshold">
+        /// The threshold for the percentage difference where the images are asumed to be the same.
+        /// The default/undefined value is <see cref="ImageComparer.DefaultImageThreshold"/>
+        /// </param>
+        /// <param name="segmentThreshold">
+        /// The threashold of the individual segments before it acumulates towards the overall difference.
+        /// The default undefined value is <see cref="ImageComparer.DefaultSegmentThreshold"/>
+        /// </param>
+        /// <param name="scalingFactor">
+        /// This is a sampling factor we sample a grid of average pixels <paramref name="scalingFactor"/> width by <paramref name="scalingFactor"/> high
+        /// The default undefined value is <see cref="ImageComparer.DefaultScalingFactor"/>
+        /// </param>
+        public static void VisualComparer<TColorA, TColorB>(Image<TColorA> expected, Image<TColorB> actual, float imageTheshold = DefaultImageThreshold, byte segmentThreshold = DefaultSegmentThreshold, int scalingFactor = DefaultScalingFactor)
```

Comment:
Shouldn't this be named differently? eg. `CheckSimilarity()`
It does an assertion!